src/Microsoft.AspNet.Razor/Parser/SyntaxTree/Block.cs

```diff
@@ -111,14 +112,18 @@ public override bool Equals(object obj)
         {
             var other = obj as Block;
             return other != null &&
-                   Type == other.Type &&
-                   Equals(CodeGenerator, other.CodeGenerator) &&
-                   ChildrenEqual(Children, other.Children);
+                Type == other.Type &&
+                Equals(CodeGenerator, other.CodeGenerator) &&
+                ChildrenEqual(Children, other.Children);
         }
 
         public override int GetHashCode()
         {
-            return (int)Type;
+            return HashCodeCombiner.Start()
+                .Add(Type)
+                .Add(CodeGenerator)
+                .Add(Children)
```

Comment:
nah, `Parent` is not an issue. `SyntaxTreeNode` is `abstract` and neither `Block` nor `Span` include it in their `Equals()` or `GetHashCode()` implementations.